### DISCUSSION TITLE
[Chapter 05] 마셜링(언마셜링)과 직렬화(역직렬화)

### DISCUSSION THREAD

**Discussion by JoisFe** [cowaine/spring-boot-for-msa#36] [Chapter 05. REST APIs with Spring MVC] [answered]
흔히 자바 객체 필드를 json 데이터형식으로 변환하는 것을 직렬화, 반대를 역질렬화 라고 많이 불렀었는데, 해당 책에서는 마셜링, 언마셜링 이라는 개념으로 표현을 하는데, 혹시 두 용어의 차이가 있나요?

**Comment by Irisation23** [comment, marked as answer]
## 직렬화와 역직렬화

- 직렬화: `객체`를 그 상태 그대로 저장하거나 전송하기 위해 `연속된 데이터로 변환`하는 것을 의미합니다. 이 데이터는 일반적으로 바이트 코드로 표현되며, 파일에 저장하거나 네트워크를 통해 전송할 수 있습니다.
  * `객체` -> `연속된 데이터 변환(일반적으로 바이트 코드)`
- 역직렬화: `직렬화된 데이터`를 다시 `객체의 형태로 변환`하는 과정을 의미합니다. 저장된 데이터나 전송된 데이터를 원래의 객체 형태로 복원하는 것입니다.
  * `연속된 데이터(일반적으로 바이트 코드)` -> `객체`

## 마샬링과 언마샬링
- 마샬링: 프로그래밍 언어나 데이터 형식에 따라 `특정 데이터 구조`를 `표준 형식으로 변환`하는 것을 의미합니다. 주로 다른 시스템 간의 데이터 교환을 위해 사용됩니다. 자바 객체를 JSON 형태로 변환하는 것도 마샬링의 한 예입니다.
  * `특정 데이터 구조` -> `표준 형식`
- 언마샬링: `마샬링된 데이터`를 `원래의 데이터 구조`로 복원하는 것을 의미합니다. JSON을 파싱하여 해당 데이터를 프로그래밍 언어의 객체로 변환하는 것도 언마샬링의 한 예입니다.
  * `마샬링 된 데이터` -> `원래의 데이터 구조`

**Reply by Irisation23** [reply]
https://wookcode.tistory.com/193

해당 참조 내용을 확인 해 보면 직렬화보다 마샬링이 보다 더 포괄적인 개념이라 할 수 있다고 하네요.
